File: fetch.ts
import { readFileSync } from "node:fs";


const config = JSON.parse(String(readFileSync("./request.json")));
const authorization = config.headers.Authorization;

console.log(authorization);

const start = performance.now();
const response = await fetch("https://71bs55wjyl.execute-api.ap-northeast-1.amazonaws.com/hello", {
  headers: {
    Authorization: authorization
  }
});

console.log(await response.text());

console.log((performance.now() - start) / 1000 + 's');

File: scripts/fetch.rs
use reqwest::Client as HttpClient;
use std::time::Instant;

#[tokio::main]
async fn main() {
    let client = HttpClient::new();
    let start = Instant::now();
    let response = client
        .get("https://p7vtih0jq1.execute-api.ap-northeast-1.amazonaws.com/hello")
        .send()
        .await
        .unwrap()
        .text()
        .await
        .unwrap();
    let duration = start.elapsed();
    print!("Duration: {duration:?}\n");
    print!("Response: {response}\n");
}
